File: DSLKIT/DLSKit/Operators/ArrayOperators.swift
//
//  ArrayOperators.swift
//  HelloWorld
//
//  Created by Pixel Logic Apps on 25/04/25.
//


import Foundation

public class ArrayOperators {
    public static func registerAll() {

        // Array.indexOf → retorna o índice de um item ou -1
        DSLOperatorRegistry.shared.register("Array.indexOf") { input, _ in
            guard let dict = input as? [String: Any],
                  let array = dict["source"] as? [Any],
                  let target = dict["search"] else {
                return nil
            }

            for (i, element) in array.enumerated() {
                if "\(element)" == "\(target)" {
                    return i
                }
            }
            return -1
        }

        // Array.contains → retorna true se o item existir
        DSLOperatorRegistry.shared.register("Array.contains") { input, _ in
            guard let dict = input as? [String: Any],
                  let array = dict["source"] as? [Any],
                  let target = dict["search"] else {
                return nil
            }

            return array.contains { "\($0)" == "\(target)" }
        }
    }
}

File: compose-dsl/src/main/java/com/example/ComposeDSL/ArrayOperators.kt
package com.example.ComposeDSL

object ArrayOperators {
    fun registerAll() {
        DSLOperatorRegistry.register("Array.indexOf") { input, _ ->
            val dict = input as? Map<*, *> ?: return@register -1
            val array = dict["source"] as? List<*> ?: return@register -1
            val search = dict["search"]
            array.indexOfFirst { "$it" == "$search" }
        }

        DSLOperatorRegistry.register("Array.contains") { input, _ ->
            val dict = input as? Map<*, *> ?: return@register false
            val array = dict["source"] as? List<*> ?: return@register false
            val search = dict["search"]
            array.any { "$it" == "$search" }
        }
    }
}
